### ISSUE TITLE
view::group_by example fails with libc++, succeeds with libstdc++

### ISSUE THREAD

**Issue by rhalbersma** [ericniebler/range-v3#162]
I can't get the calendar example to compile on master trunk (clang 3.5 / 3.6, [Coliru online](http://coliru.stacked-crooked.com/a/c01c6567b47460e1)). The CMakeLists.txt mentions that there are some fixes pending on develop. Could they be merged into master? 

I tried debugging with a [simplified calendar example](http://coliru.stacked-crooked.com/a/8d48beadedac32b0) that strangely works online, but not on master trunk. Seems something is off with `group_by`. What's happening here?

**Comment by ericniebler** [comment]
I had to make fixes to Boost.Range. I'm not sure if they have been merged
to master yet.

**Comment by rhalbersma** [comment]
Ah, *Boost* Range was the culprit. Now I see it in the error message. That's rather nasty to be pulled in like that.

BTW, the simplified calendar example I mentiond earlier (that emulates your slides 19/20 from CppNow) compiles with [clang 3.6.0 and libstdc++](http://coliru.stacked-crooked.com/a/8d48beadedac32b0), but [not with clang 3.6.0 and libc++](http://coliru.stacked-crooked.com/a/8a161b8693bf2d2e). The error comes from a failed `IndirectCallablePredicate` inside `view::group_by` that brances of into libc++'s `__invoke()`.

**Comment by ericniebler** [comment]
Have you sync'ed libc++ recently? That sounds like a bug I reported and has
since been fixed.

**Comment by rhalbersma** [comment]
fixed :)
